src/dsp_tools/commands/xmlupload/xmlupload.py

```diff
@@ -123,6 +129,16 @@ def xmlupload(
     return cleanup_upload(upload_state)
 
 
+def _validate_iiif_uri(root: etree._Element) -> None:
+    uris = [node.text for node in root.iter(tag="iiif-uri")]
+    all_uris = cast(list[str], uris)
+    problems = [res for x in all_uris if (res := IIIFUriValidatorLive(x, is_iiif_uri(x)).validate())]
```

Comment:
I have thought about that option, but I did not want to mix the functionality. It should get the result if the validation passed, because then the error message is constructed best.